gst/nnstreamer/tensor_filter/tensor_filter_single.c

```diff
@@ -242,27 +244,48 @@ g_tensor_filter_allocate_in_invoke (GTensorFilterSingle * self)
   return spriv->allocate_in_invoke;
 }
 
+/**
+ * @brief Internal callback for async invoke.
+ */
+static int
+g_tensor_filter_single_async_cb (GstTensorMemory *data, GstTensorsInfo *info,
+    void *user_data)
+{
+  GTensorFilterSingle *self;
+  GTensorFilterSinglePrivate *spriv;
+  int ret = 0;
+
+  self = G_TENSOR_FILTER_SINGLE (user_data);
+  spriv = G_TENSOR_FILTER_SINGLE_PRIV (self);
+
+  if (spriv->async_cb) {
+    ret = spriv->async_cb (data, info, spriv->async_pdata);
```

Comment:
How about checking for null before using the data?

Reply:
updated